LookAndFeels/foleys_Skeuomorphic.cpp

```diff
@@ -0,0 +1,240 @@
+                                             { 1.0,                          blackA092 } });
+    fillEllipse (g, bounds, backgroundGr);
+
+
+
+    bounds.reduce (diameter * 0.15f, diameter * 0.15f);
+    float fgRadius = bounds.getWidth() * 0.5f;
+
+    Image fg = Image (Image::ARGB, diameter, diameter, true);
+
+    Graphics gg (fg); // lower shadow
+    {
+        float xOffset = fgRadius * 0.12f;
+        float yOffset = fgRadius * 0.32f;
+
+        ColourGradient shadowGr = makeGradient ({ centre.x + xOffset, centre.y + yOffset },
+                                                { centre.x, centre.y + fgRadius + yOffset }, true,
+                                             {  { 0.0,  blackA122                 },
+                                                { 0.72, blackA122                 },
+                                                { 1.0,  Colours::transparentBlack }  });
+        fillEllipse (gg, bounds.translated (xOffset, yOffset), shadowGr);
+    }
+
+    // middle
+    {
+        gg.saveState();
+        clipToEllipse (gg, bounds);
+
+        ColourGradient gr { { 55, 55, 66 }, 0.0f, bounds.getY(),
+                            { 34, 35, 41 }, 0.0f, bounds.getBottom(), false };
+        fillEllipse (gg, bounds, gr);
+
+        ColourGradient ringGr = makeGradient (centre, { centre.x, centre.y + fgRadius }, true,
+                                           {  { 0.0,  Colours::transparentBlack },
+                                              { 0.72, Colours::transparentBlack },
+                                              { 1.0,  blackA042                 }  });
+        fillEllipse (gg, bounds, ringGr);
+
+        gg.setColour (whiteA042);
+        gg.drawEllipse (bounds.translated (0.0f, 1.0f), 1.0f);
+        if (!isSmall) gg.drawEllipse (bounds.translated (0.0f, 2.0f), 1.0f);
+
+        gg.restoreState();
+        gg.setColour (blackA092);
+        gg.drawEllipse (bounds, 1.0f);
+    }
+
+    knobsBgs.insert ({diameter, knobImages { std::move (bg), std::move (fg) }});
+    return knobsBgs.find (diameter)->second;
+}
+
+void Skeuomorphic::drawRotarySlider (Graphics& g, int x, int y, int width, int height, float sliderPos,
+                                     const float rotaryStartAngle, const float rotaryEndAngle, Slider& slider)
+{
+    int diameter = (width > height)? height : width;
+    if (diameter < 16) return;
+
+    Point<float> centre (x + int (width * 0.5f + 0.5f), y + int (height * 0.5f + 0.5f));
+    diameter -= (diameter % 2)? 9 : 8;
+    float radius = diameter * 0.5f;
+    x = int (centre.x - radius);
+    y = int (centre.y - radius);
+
+    Rectangle<float> bounds (x, y, diameter, diameter);
+
+    bool isTiny  = diameter < 20;
+    bool isSmall = diameter < 60;
+
+    const auto toAngle = rotaryStartAngle + sliderPos * (rotaryEndAngle - rotaryStartAngle);
+
+//    Using only the fill color for now
+//    const Colour outline = slider.findColour (Slider::rotarySliderOutlineColourId);
```

Comment:
make a decision: fix it or remove it ;-)

Reply:
removed it :D